elastic/beats
# [Metricbeat] Fix cloudwatch metricset missing tags collection

## What does this PR do?

This PR is to fix cloudwatch metricset from missing tags when `tags.resource_type_filter` is given. When user tries to collect daily storage metrics and s3 bucket tags with the cloudwatch metricset, it will fail to collect tags. This is because for CloudWatch metrics from S3 buckets, they always have `StorageType` as dimension. Dimension and S3 bucket name are combined to be the `identifierValue` in cloudwatch metricset code. But when collecting tags, dimension `StorageType` is not collected. That's why there is a mismatch between `identifierValue` and `resourceTagMap`.

## Why is it important?

This PR will fix tags collection for all namespaces that always have a dimension reported in CloudWatch metrics.

## Checklist


## How to test this PR locally
1. Make sure the AWS account you use for testing has S3 buckets with tags.
2. Use config below for aws.yml:
```
- module: aws
  period: 86400s
  metricsets:
    - cloudwatch
  metrics:
    - namespace: AWS/S3
      tags.resource_type_filter: s3
```
3. After starting Metricbeat, you should see tags collected from S3 buckets.

## Related issues

- Closes https://github.com/elastic/beats/issues/17419
